### DISCUSSION TITLE
skynet开发mmo场景内心跳问题

### DISCUSSION THREAD

**Discussion by ghost** [cloudwu/skynet#1810] [Q&A] [answered]
大家开发mmo，场景内帧间隔一般设置多久一跳。我目前0.1秒一跳感觉有点浪费性能

**Comment by BanKnight** [comment]
你这个问题有点粗略。
如果是跑npc等ai，这个是要跟着需求走的，如果单位多的话，还要想办法做分组，降低频次。
普通timer，我这里一般就是1s了

**Reply by Gowa2017** [reply]
- -你想说的心跳是什么东西？
帧间隔？60帧？

**Reply by ghost** [reply]
是的帧间隔

**Comment by cloudwu** [comment, marked as answer]
对于服务器来说，真实时间 walltime 是没有太大意义的。有意义的是事件的发生次序。
事件包括了业务本身产生的动作，以及玩家外部的输入。

鉴于外部输入必然有一定的网络延迟，你可以设置一个下限，比如 100ms ，即，即使玩家可以用少于 100ms 把消息发送到服务器，你也认为它至少要 100ms 。

然后，服务器要做的事情就是，把所有的事件打上时间戳，然后根据时间戳的次序依次处理这些事件，并不关心真正处理它们时的 walltime 。而超过 100ms 的事件都可以堆积在一起放在待处理集合中。

所以，服务器流程就是：
1. 如果有 100ms 以内的事件发生，那么就按次序执行它们，直到这个队列为空。
2. 对长期事件做排序，直到它们变成短期事件。
3. 并不需要周期性的运行某个特定业务，需要的只是处理事件。

这个流程中并不需要太高频和太准确（符合 walltime）的 timer 支持。
